Lib/py_compile.py

```diff
@@ -174,44 +175,44 @@ def compile(file, cfile=None, dfile=None, doraise=False, optimize=-1,
 
 
 def main(args=None):
-    """Compile several source files.
-
-    The files named in 'args' (or on the command line, if 'args' is
-    not specified) are compiled and the resulting bytecode is cached
-    in the normal manner.  This function does not search a directory
-    structure to locate source files; it only compiles files named
-    explicitly.  If '-' is the only parameter in args, the list of
-    files is taken from standard input.
-
-    """
+    """Script main program."""
     if args is None:
         args = sys.argv[1:]
+    parser = argparse.ArgumentParser(
+        prog=__name__,
+        description=(
+            "Compiles several source files. The files named "
+            "in 'filenames' are compiled and the resulting "
+            "bytecode is cached in the normal manner. This "
+            "program does not search a directory structure "
+            "to locate source files; it only compiles files "
+            "named explicitly. If '-' is the only file name "
+            "in 'filenames', the list of files taken from "
+            "standard input. If flag '-q/--quiet' is set "
+            "errors output would be suppressed."))
+    parser.add_argument(
+        '-q', '--quiet', default=False,
+        action='store_true', help='suppress errors output')
+    parser.add_argument(
+        'filenames', nargs='+', help='files to compile')
+    args = parser.parse_args(args)
     rv = 0
-    if args == ['-']:
-        while True:
-            filename = sys.stdin.readline()
-            if not filename:
-                break
-            filename = filename.rstrip('\n')
-            try:
-                compile(filename, doraise=True)
-            except PyCompileError as error:
-                rv = 1
-                if quiet < 2:
-                    sys.stderr.write("%s\n" % error.msg)
-            except OSError as error:
-                rv = 1
-                if quiet < 2:
-                    sys.stderr.write("%s\n" % error)
+    if args.filenames == ['-']:
+        filenames = (line.rstrip('\n') for line in sys.stdin)
     else:
-        for filename in args:
-            try:
-                compile(filename, doraise=True)
-            except PyCompileError as error:
-                # return value to indicate at least one failure
-                rv = 1
-                if quiet < 2:
-                    sys.stderr.write("%s\n" % error.msg)
+        filenames = args.filenames
+    for filename in filenames:
+        try:
+            compile(filename, doraise=True)
+        except PyCompileError as error:
+            # return value to indicate at least one failure
+            rv = 1
+            if not args.quiet:
+                sys.stderr.write("%s\n" % error.msg)
```

Comment:
```suggestion
                print(error.msg, file=sys.stderr)
```
Print function is super!

Reply:
agree with you